crowbar_framework/app/models/node_object.rb

```diff
@@ -659,7 +659,12 @@ def rebuild_run_list
     crowbar["run_list_map"] = {} if crowbar["run_list_map"].nil?
 
     # Cull by state
-    map = crowbar["run_list_map"].select { |k,v| v["states"].include?("all") or v["states"].include?(self.crowbar['state']) }
+    map = crowbar["run_list_map"].select do |k, v|
+      v["states"].include?(state) ||
```

Comment:
Actually, I think the move from `self.crowbar['state']` to `state` is wrong: we should never do something depending on `noupdate`, imho.